engine/script/src/script_timer.cpp

```diff
@@ -323,7 +323,7 @@ namespace dmScript
         }
 
         timer.m_IsAlive = 0;
-        timer.m_Callback(timer_world, TIMER_EVENT_CANCELLED, timer.m_Handle, 0.f, timer.m_Owner, timer.m_UserData);
+        timer.m_Callback(timer_world, finish ? TIMER_EVENT_FINISHED : TIMER_EVENT_CANCELLED, timer.m_Handle, 0.f, timer.m_Owner, timer.m_UserData);
```

Comment:
Wait now...

Isn't the callback _already_ being called upon its cancellation?
Isn't that enough?

The main premise of the issue:
"Sometimes you may want to cancel the timer from counting down but still run any callback it may have inside of it."

So, it already has the workings of doing a callback.
I think a better solution is to pass the boolean through to the `LuaTimerCallback`

Otherwise, it's the same issue as above, we call it a "finish" when in fact it's the "cancel" event. Although it might work in this case, it's a problematic programming pattern to have, and we want to set good examples in the code.